### ISSUE TITLE
CovariantEqualsCheck: covariant equals on top-level method not flagged in compact source files (JEP 512)

### ISSUE THREAD

**Issue by vivek-0509** [checkstyle/checkstyle#20552]
Parent Issue #19971

I have read check documentation: https://checkstyle.org/checks/coding/covariantequals.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > config.xml <<'EOF'
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="CovariantEquals"/>
  </module>
</module>
EOF

vivek@Viveks-MacBook-Air checkstyle % cat > InputCovariantEqualsCompact.java <<'EOF'
public boolean equals(String other) {
    return false;
}

void main() { }
EOF

vivek@Viveks-MacBook-Air checkstyle % RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar target/checkstyle-13.8.0-SNAPSHOT-all.jar -c config.xml InputCovariantEqualsCompact.java

Starting audit...
Audit done.
```

---

**Describe what you expect in detail.**

CovariantEqualsCheck must flag a covariant `equals` method declared at the top level of a compact source file. Under JEP 512 the implicit class is a real class that inherits from `Object`, so a `boolean equals(String)` declared next to no `boolean equals(Object)` override is exactly the design mistake the check exists to catch. The same code inside an ordinary class is flagged with `covariant equals without overriding equals(java.lang.Object).`

Expected on the compact input: one violation on line 1: `covariant equals without overriding equals(java.lang.Object). [CovariantEquals]`.

Root cause: `CovariantEqualsCheck` registers for `CLASS_DEF`, `ENUM_DEF`, `RECORD_DEF`, `LITERAL_NEW` and walks each one's body to find `equals` methods. It does not register for `COMPACT_COMPILATION_UNIT`. In a compact source file the top-level methods are members of the implicit class, but the check never enters an implicit-class context because there is no `CLASS_DEF` for it. The `equals(String)` is silently ignored.

Fix: also register for `COMPACT_COMPILATION_UNIT` and apply the same body-walk logic to its top-level `METHOD_DEF` children.

**Same code inside an ordinary class:**

```bash
vivek@Viveks-MacBook-Air checkstyle % cat > InputCovariantEqualsOrdinary.java <<'EOF'
public class InputCovariantEqualsOrdinary {
    public boolean equals(String other) {
        return false;
    }

    public void main() { }
}
EOF

vivek@Viveks-MacBook-Air checkstyle % java $RUN_LOCALE -jar target/checkstyle-13.8.0-SNAPSHOT-all.jar -c config.xml InputCovariantEqualsOrdinary.java

Starting audit...
[ERROR] InputCovariantEqualsOrdinary.java:2:20: covariant equals without overriding equals(java.lang.Object). [CovariantEquals]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by Carbon14-48** [comment]
on it
